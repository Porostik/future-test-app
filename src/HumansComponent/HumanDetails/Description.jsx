import React from 'react';

function Description({ title, value }) {
  return (
    <div>
      <span className="title">{title}</span>
      <textarea className="description" value={value} disabled />
    </div>
  );
}

export default Description;
