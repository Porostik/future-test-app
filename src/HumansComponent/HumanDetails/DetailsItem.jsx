import React from 'react';

// Компонент строки данных (заголовк + значение)
function DetailsItem({ title, value }) {
  return (
    <div className="details-item">
      <span className="title">{title}</span>
      <p className="value">{value}</p>
    </div>
  );
}

export default DetailsItem;
