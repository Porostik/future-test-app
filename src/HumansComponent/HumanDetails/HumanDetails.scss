.details {
  grid-column: 2;
  height: 400px;
  padding: 20px;
  align-self: start;
  background-color: #efdb9d;
  border-radius: 10px;
  color: #5d5843;
  font-family: 'Raleway', sans-serif;
  display: flex;
  justify-content: space-between;
  flex-direction: column;

  .title,
  .value {
    background-color: #b6be82;
    padding: 5px;
    border-radius: 5px;
  }

  .value {
    font-weight: 800;
  }
}

.description {
  width: 100%;
  height: 140px;
  resize: none;
  padding: 5px 5px 10px 5px;
  border: none;
  background-color: #b6be82;
  border-radius: 5px;
  font-size: 16px;
  margin-top: 15px;
}

.details-item {
  display: flex;
  justify-content: space-between;
  font-size: 16px;
}
