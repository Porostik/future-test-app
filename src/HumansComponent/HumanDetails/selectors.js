import { createSelector } from 'reselect';
import { createDetailsObj } from 'utils';
import { detailsDefaultProps } from 'constants.js';

const detailsObj = (store) => store.openDetailsItem;

export const selectDetailsItem = createSelector([detailsObj], (item) => {
  //   console.log(item);
  //   console.log(detailsDefaultProps);
  return createDetailsObj(item, detailsDefaultProps);
});
