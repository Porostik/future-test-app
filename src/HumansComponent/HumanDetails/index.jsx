import React from 'react';
import './HumanDetails.scss';
import { useSelector } from 'react-redux';

import { detailsTitles } from 'constants.js';
import Description from './Description';
import DetailsItem from './DetailsItem';
import { selectDetailsItem } from './selectors';

// Компонент отображения детальных данных о человеке
function HumanDetails() {
  const humanDetails = useSelector(selectDetailsItem);

  return (
    humanDetails && (
      <div className="details">
        <DetailsItem
          title={detailsTitles.fullName}
          value={`${humanDetails.firstName} ${humanDetails.lastName}`}
        />
        <Description title={detailsTitles.description} value={humanDetails.description} />
        {Object.keys(humanDetails.address).map((item, index) => (
          <DetailsItem
            key={`${item}_${index}`}
            title={detailsTitles[item]}
            value={humanDetails.address[item]}
          />
        ))}
      </div>
    )
  );
}

export default HumanDetails;
