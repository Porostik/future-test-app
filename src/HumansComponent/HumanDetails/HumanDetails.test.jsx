import React from 'react';
import { mount } from 'enzyme';
import HumanDetails from './index';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('Filter component', () => {
  test('should render human details from redux', () => {
    const store = mockStore({
      openDetailsItem: {
        id: 101,
        firstName: 'Sue',
        lastName: 'Corson',
        email: 'DWhalley@in.gov',
        phone: '(612)211-6296',
        address: {
          streetAddress: '9792 Mattis Ct',
          city: 'Waukesha',
          state: 'WI',
          zip: '22178',
        },
        description: 'et lacus magna dolor...',
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <HumanDetails />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
