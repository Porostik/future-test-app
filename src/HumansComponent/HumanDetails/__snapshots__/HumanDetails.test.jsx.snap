// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render human details from redux 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HumanDetails>
    <div
      className="details"
    >
      <DetailsItem
        title="Выбран пользователь"
        value="Sue Corson"
      >
        <div
          className="details-item"
        >
          <span
            className="title"
          >
            Выбран пользователь
          </span>
          <p
            className="value"
          >
            Sue Corson
          </p>
        </div>
      </DetailsItem>
      <Description
        title="Описание:"
        value="et lacus magna dolor..."
      >
        <div>
          <span
            className="title"
          >
            Описание:
          </span>
          <textarea
            className="description"
            disabled={true}
            value="et lacus magna dolor..."
          />
        </div>
      </Description>
      <DetailsItem
        key="streetAddress_0"
        title="Адрес проживания:"
        value="9792 Mattis Ct"
      >
        <div
          className="details-item"
        >
          <span
            className="title"
          >
            Адрес проживания:
          </span>
          <p
            className="value"
          >
            9792 Mattis Ct
          </p>
        </div>
      </DetailsItem>
      <DetailsItem
        key="city_1"
        title="Город:"
        value="Waukesha"
      >
        <div
          className="details-item"
        >
          <span
            className="title"
          >
            Город:
          </span>
          <p
            className="value"
          >
            Waukesha
          </p>
        </div>
      </DetailsItem>
      <DetailsItem
        key="state_2"
        title="Провинция/штат:"
        value="WI"
      >
        <div
          className="details-item"
        >
          <span
            className="title"
          >
            Провинция/штат:
          </span>
          <p
            className="value"
          >
            WI
          </p>
        </div>
      </DetailsItem>
      <DetailsItem
        key="zip_3"
        title="Индекс:"
        value="22178"
      >
        <div
          className="details-item"
        >
          <span
            className="title"
          >
            Индекс:
          </span>
          <p
            className="value"
          >
            22178
          </p>
        </div>
      </DetailsItem>
    </div>
  </HumanDetails>
</Provider>
`;
